### PR TITLE
build(build-tools): Upgrade oclif dependencies

### PR DESCRIPTION
**Pull request by tylerbutler**
Upgrades oclif-related dependencies to the latest version. The most important upgrade is to @oclif/test, which was a major version behind latest because of a test issue that was addressed in the latest patch.

### CODE REVIEW

File: build-tools/pnpm-lock.yaml [hunk 1]
```diff
@@ -1565,7 +1924,7 @@ packages:
       - typescript
     dev: true
 
-  /@fluid-tools/build-cli/0.29.0_typescript@5.1.6:
+  /@fluid-tools/build-cli/0.29.0_lazgpcpm5xssvy6zj7dsqbusoe:
```
Comment (jason-ha) [inline build-tools/pnpm-lock.yaml:1927]: Do you know why this is going to a hash and not longer a named distinguisher?

File: build-tools/pnpm-lock.yaml [hunk 2]
```diff
@@ -11897,6 +12440,13 @@ packages:
     requiresBuild: true
     dev: true
 
+  /yarn/1.22.22:
+    resolution: {integrity: sha512-prL3kGtyG7o9Z9Sv8IPfBNrWTDmXB4Qbes8A9rEzt6wkJV8mUvoirjU0Mp3GGAU06Y0XQyA3/2/RQFVuK7MTfg==}
+    engines: {node: '>=4.0.0'}
+    hasBin: true
+    requiresBuild: true
+    dev: false
+
```
Comment (jason-ha) [inline build-tools/pnpm-lock.yaml:12449]: Something now pulls in yarn? For what?

File: build-tools/pnpm-lock.yaml [hunk 3]
```diff
@@ -710,6 +711,53 @@ packages:
     transitivePeerDependencies:
       - aws-crt
 
+  /@aws-sdk/client-sso-oidc/3.568.0_f5hjed2pgq3vcatdx6pdivrwea:
```
Comment (jason-ha) [inline build-tools/pnpm-lock.yaml:714]: no good deduplication with these big **non-dev** aws additions?

**Review by jason-ha — APPROVED** [review]
Some questions about dep changes, but can't say they have to be addressed.

File: build-tools/pnpm-lock.yaml (hunk 2 above)
Comment (alexvy86) [reply build-tools/pnpm-lock.yaml:12449]: The new patch of @oclif/plugins-plugins

image

[From a previous patch](https://github.com/oclif/plugin-plugins/releases/tag/5.0.11). Seems reasonable based on what oclif is...?

**Review by alexvy86 — APPROVED** [review]

File: build-tools/pnpm-lock.yaml (hunk 2 above)
Comment (tylerbutler) [reply build-tools/pnpm-lock.yaml:12449]: It's part of the default oclif setup, but we don't currently use it. I'll remove the dep and the config in package.json. We can always add it back if we want to use the feature later.

File: build-tools/pnpm-lock.yaml [hunk 4]
```diff
@@ -1022,6 +1212,16 @@ packages:
...
+  /@aws-sdk/middleware-user-agent/3.567.0:
+    resolution: {integrity: sha512-a7DBGMRBLWJU3BqrQjOtKS4/RcCh/BhhKqwjCE0FEhhm6A/GGuAs/DcBGOl6Y8Wfsby3vejSlppTLH/qtV1E9w==}
+    engines: {node: '>=16.0.0'}
+    dependencies:
+      '@aws-sdk/types': 3.567.0
+      '@aws-sdk/util-endpoints': 3.567.0
+      '@smithy/protocol-http': 3.3.0
+      '@smithy/types': 2.12.0
+      tslib: 2.6.2
+
```
Comment (jason-ha) [inline build-tools/pnpm-lock.yaml:1224]: I still see `@aws-sdk` duplication like this. Were recent changes meant to address these cases? Should we have any `@aws-sdk` in here?

**Pull request merged**